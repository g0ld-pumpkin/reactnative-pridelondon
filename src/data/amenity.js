// @flow
import * as decode from "../lib/decode";
import type { Decoder } from "../lib/decode";

type AmenityType = "Toilet" | "First Aid";

const amenityTypeDecoder: Decoder<AmenityType> = decode.oneOf([
  decode.value("Toilet"),
  decode.value("First Aid")
]);

export type Amenity = {
  // important to keep this at the top level so type refinement works
  contentType: "amenity",
  id: string,
  locale: string,
  revision: number,
  fields: {
    type: AmenityType,
    location: { lat: number, lon: number }
  }
};

const decodeAmenity = (locale: string): Decoder<Amenity> =>
  decode.shape({
    contentType: decode.at(
      ["sys", "contentType", "sys", "id"],
      decode.value("amenity")
    ),
    id: decode.at(["sys", "id"], decode.string),
    locale: decode.succeed(locale),
    revision: decode.at(["sys", "revision"], decode.number),
    fields: decode.field(
      "fields",
      decode.shape({
        type: decode.at(["type", locale], amenityTypeDecoder),
        location: decode.at(
          ["location", locale],
          decode.shape({
            lat: decode.field("lat", decode.number),
            lon: decode.field("lon", decode.number)
          })
        )
      })
    )
  });

export default decodeAmenity;
