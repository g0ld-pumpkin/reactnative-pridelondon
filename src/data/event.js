// @flow
import R from "ramda";
import {
  set as setDate,
  diff as diffDate,
  add as addToDate,
  toFormat as formatDate,
  FORMAT_EUROPEAN_DATE,
  FORMAT_CONTENTFUL_ISO
} from "../lib/date";
import type { Maybe } from "../lib/maybe";
import * as maybe from "../lib/maybe";
import type { Decoder } from "../lib/decode";
import * as decode from "../lib/decode";
import type { FieldRef } from "./field-ref";
import decodeFieldRef from "./field-ref";
import * as colors from "../constants/colors";

export type Events = {
  [id: string]: Event
};

export type EventCategoryName =
  | "Cabaret and Variety"
  | "Community"
  | "Talks and Debates"
  | "Film and Screenings"
  | "Plays and Theatre"
  | "Social and Networking"
  | "Nightlife"
  | "Exhibition and Tours"
  | "Sports and Activities"
  | "Health"
  | "Music";

export const eventCategoryNames = [
  "Cabaret and Variety",
  "Community",
  "Talks and Debates",
  "Film and Screenings",
  "Plays and Theatre",
  "Social and Networking",
  "Nightlife",
  "Exhibition and Tours",
  "Sports and Activities",
  "Health",
  "Music"
];

const decodeEventCategoryName = decode.oneOf(
  eventCategoryNames.map(decode.value)
);

export type EventCategory = {
  label: EventCategoryName,
  color: string,
  contrast: boolean
};

// We use a switch here, rather than a look up so that we can ensure that
// this function will always return an EventCategory.
export const getEventCategoryFromName = (
  category: EventCategoryName
): EventCategory => {
  switch (category) {
    case "Cabaret and Variety":
      return {
        label: "Cabaret and Variety",
        color: colors.coralColor,
        contrast: false
      };
    case "Community":
      return {
        label: "Community",
        color: colors.brightLightBlueColor,
        contrast: true
      };
    case "Talks and Debates":
      return {
        label: "Talks and Debates",
        color: colors.darkSkyBlueColor,
        contrast: false
      };
    case "Film and Screenings":
      return {
        label: "Film and Screenings",
        color: colors.lightTealColor,
        contrast: true
      };
    case "Plays and Theatre":
      return {
        label: "Plays and Theatre",
        color: colors.warmPinkColor,
        contrast: false
      };
    case "Social and Networking":
      return {
        label: "Social and Networking",
        color: colors.turquoiseBlueColor,
        contrast: true
      };
    case "Nightlife":
      return {
        label: "Nightlife",
        color: colors.yellowColor,
        contrast: true
      };
    case "Exhibition and Tours":
      return {
        label: "Exhibition and Tours",
        color: colors.brightPurpleColor,
        contrast: false
      };
    case "Sports and Activities":
      return {
        label: "Sports and Activities",
        color: colors.vomitYellowColor,
        contrast: true
      };
    case "Health":
      return {
        label: "Health",
        color: colors.bubblegumPinkColor,
        contrast: true
      };
    case "Music":
      return {
        label: "Music",
        color: colors.cornflowerBlueColor,
        contrast: false
      };
    default:
      // Default case with cast to enable flow exhaustive match. This will
      // cause flow to error if passed anything other than an EventCategory;
      return (category: empty);
  }
};

export type Event = {
  // important to keep this at the top level so type refinement works
  contentType: "event",
  id: string,
  locale: string,
  revision: number,
  fields: {
    name: string,
    eventCategories: Array<EventCategoryName>,
    audience: Array<string>,
    startTime: string,
    endTime: string,
    location: { lat: number, lon: number },
    addressLine1: Maybe<string>,
    addressLine2: Maybe<string>,
    city: Maybe<string>,
    postcode: Maybe<string>,
    locationName: string,
    eventPriceLow: number,
    eventPriceHigh: number,
    accessibilityOptions: Array<string>,
    eventDescription: string,
    accessibilityDetails: Maybe<string>,
    email: Maybe<string>,
    phone: Maybe<string>,
    ticketingUrl: Maybe<string>,
    venueDetails: Array<string>,
    individualEventPicture: FieldRef,
    eventsListPicture: FieldRef,
    performances: Array<FieldRef>,
    recurrenceDates: Array<string>,
    stage: boolean
  }
};

export type SavedEvents = Set<string>;

export type EventDays = Event[][];

const maybeField = <A>(
  locale: string,
  field: string,
  decoder: Decoder<A>
): Decoder<Maybe<A>> =>
  decode.field(field, decode.maybe(decode.field(locale, decoder)));

const maybeFieldWithDefault = <A>(
  locale: string,
  field: string,
  decoder: Decoder<A>,
  defaultValue: A
): Decoder<A> =>
  decode.map(
    maybe.withDefault(defaultValue),
    maybeField(locale, field, decoder)
  );

export const decodeEvent = (locale: string): Decoder<Event> =>
  decode.shape({
    contentType: decode.at(
      ["sys", "contentType", "sys", "id"],
      decode.value("event")
    ),
    id: decode.at(["sys", "id"], decode.string),
    locale: decode.succeed(locale),
    revision: decode.at(["sys", "revision"], decode.number),
    fields: decode.field(
      "fields",
      decode.shape({
        name: decode.at(["name", locale], decode.string),
        eventCategories: decode.at(
          ["eventCategories", locale],
          decode.array(decodeEventCategoryName)
        ),
        audience: maybeFieldWithDefault(
          locale,
          "audience",
          decode.array(decode.string),
          []
        ),
        // may want to combine startTime and endTime to fix the
        // date flipping issue in the decoder
        // may also want to convert it to a DateTime
        startTime: decode.at(["startTime", locale], decode.string),
        endTime: decode.at(["endTime", locale], decode.string),
        location: decode.at(
          ["location", locale],
          decode.shape({
            lat: decode.field("lat", decode.number),
            lon: decode.field("lon", decode.number)
          })
        ),
        addressLine1: maybeField(locale, "addressLine1", decode.string),
        addressLine2: maybeField(locale, "addressLine2", decode.string),
        city: maybeField(locale, "city", decode.string),
        postcode: maybeField(locale, "postcode", decode.string),
        locationName: decode.at(["locationName", locale], decode.string),
        eventPriceLow: decode.at(["eventPriceLow", locale], decode.number),
        eventPriceHigh: decode.at(["eventPriceHigh", locale], decode.number),
        accessibilityOptions: maybeFieldWithDefault(
          locale,
          "accessibilityOptions",
          decode.array(decode.string),
          []
        ),
        eventDescription: decode.at(
          ["eventDescription", locale],
          decode.string
        ),
        accessibilityDetails: maybeField(
          locale,
          "accessibilityDetails",
          decode.string
        ),
        email: maybeField(locale, "email", decode.string),
        phone: maybeField(locale, "phone", decode.string),
        ticketingUrl: maybeField(locale, "ticketingUrl", decode.string),
        venueDetails: maybeFieldWithDefault(
          locale,
          "venueDetails",
          decode.array(decode.string),
          []
        ),
        individualEventPicture: decode.at(
          ["individualEventPicture", locale],
          decodeFieldRef
        ),
        eventsListPicture: decode.at(
          ["eventsListPicture", locale],
          decodeFieldRef
        ),
        performances: maybeFieldWithDefault(
          locale,
          "performances",
          decode.array(decodeFieldRef),
          []
        ),
        recurrenceDates: maybeFieldWithDefault(
          locale,
          "recurrenceDates",
          decode.array(decode.string),
          []
        ),
        stage: maybeFieldWithDefault(locale, "stage", decode.boolean, false)
      })
    )
  });

const formatEuropeanDate = value => formatDate(value, FORMAT_EUROPEAN_DATE);

const generateRecurringEvent = (event: Event) => (
  recurrenceStartTime: string
) => {
  const { endTime, startTime } = event.fields;

  const difference = diffDate(recurrenceStartTime, startTime);
  const recurrenceEndTime = addToDate(endTime, difference);

  return R.mergeDeepRight(event, {
    fields: {
      startTime: formatDate(recurrenceStartTime, FORMAT_CONTENTFUL_ISO),
      endTime: formatDate(recurrenceEndTime, FORMAT_CONTENTFUL_ISO),
      recurrenceDates: [
        formatEuropeanDate(startTime),
        ...event.fields.recurrenceDates
      ]
    },
    id: `${event.id}-recurrence-${formatEuropeanDate(recurrenceStartTime)}`
  });
};

const recurrenceDateToStartTime = (originalStartTime: string) => (
  recurrence: string
) => {
  const [recurrenceDay, recurrenceMonth, recurrenceYear] = recurrence.split(
    "/"
  );

  return formatDate(
    setDate(originalStartTime, {
      year: recurrenceYear,
      month: recurrenceMonth,
      day: recurrenceDay
    }),
    FORMAT_CONTENTFUL_ISO
  );
};

export const expandRecurringEvents = (event: Event): Array<Event> => {
  const recurrenceStartTimes = [
    event.fields.startTime,
    ...event.fields.recurrenceDates.map(
      recurrenceDateToStartTime(event.fields.startTime)
    )
  ];

  const events = R.uniq(recurrenceStartTimes)
    .slice(1)
    .map(generateRecurringEvent(event));
  return [event, ...events];
};
