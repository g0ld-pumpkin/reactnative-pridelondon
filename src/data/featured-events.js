// @flow
import * as decode from "../lib/decode";
import type { Decoder } from "../lib/decode";
import type { FieldRef } from "./field-ref";
import decodeFieldRef from "./field-ref";

export type FeaturedEvents = {
  // important to keep this at the top level so type refinement works
  contentType: "featuredEvents",
  id: string,
  locale: string,
  revision: number,
  fields: {
    title: string,
    events: Array<FieldRef>
  }
};

const decodeFeaturedEvents = (locale: string): Decoder<FeaturedEvents> =>
  decode.shape({
    contentType: decode.at(
      ["sys", "contentType", "sys", "id"],
      decode.value("featuredEvents")
    ),
    id: decode.at(["sys", "id"], decode.string),
    locale: decode.succeed(locale),
    revision: decode.at(["sys", "revision"], decode.number),
    fields: decode.field(
      "fields",
      decode.shape({
        title: decode.at(["title", locale], decode.string),
        events: decode.at(["events", locale], decode.array(decodeFieldRef))
      })
    )
  });

export default decodeFeaturedEvents;
