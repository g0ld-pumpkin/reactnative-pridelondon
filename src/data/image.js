// @flow
import * as decode from "../lib/decode";
import type { Decoder } from "../lib/decode";

export type Images = {
  [id: string]: ImageDetails
};

export type ImageDetails = {
  id: string,
  revision: number,
  uri: string,
  width: number,
  height: number
};

export const getImageDetails = (images: Images) => (
  id: string,
  dimensions: ?{ width: number, height: number }
): ?ImageDetails => {
  const imageDetails = images[id];

  // Image details might be undefined in case of unpublished image
  if (!dimensions || !imageDetails) {
    return imageDetails;
  }

  const imageUriWithResizingBehaviourParameters = `${imageDetails.uri}?w=${
    dimensions.width
  }&h=${dimensions.height}&fit=fill`;
  return Object.assign({}, imageDetails, {
    uri: imageUriWithResizingBehaviourParameters
  });
};

export const decodeImageDetails = (locale: string): Decoder<ImageDetails> =>
  decode.shape({
    id: decode.at(["sys", "id"], decode.string),
    revision: decode.at(["sys", "revision"], decode.number),
    uri: decode.map(
      value => `https:${value}`,
      decode.at(["fields", "file", locale, "url"], decode.string)
    ),
    height: decode.at(
      ["fields", "file", locale, "details", "image", "height"],
      decode.number
    ),
    width: decode.at(
      ["fields", "file", locale, "details", "image", "width"],
      decode.number
    )
  });
