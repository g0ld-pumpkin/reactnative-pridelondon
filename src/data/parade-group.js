// @flow
import * as decode from "../lib/decode";
import type { Decoder } from "../lib/decode";
import type { Maybe } from "../lib/maybe";

export type ParadeGroup = {
  // important to keep this at the top level so type refinement works
  contentType: "paradeGroup",
  id: string,
  locale: string,
  revision: number,
  fields: {
    name: string,
    facebookUrl: Maybe<string>,
    twitterUrl: Maybe<string>,
    websiteUrl: Maybe<string>
  }
};

const maybeField = <A>(
  locale: string,
  field: string,
  decoder: Decoder<A>
): Decoder<Maybe<A>> =>
  decode.field(field, decode.maybe(decode.field(locale, decoder)));

const decodeParadeGroup = (locale: string): Decoder<ParadeGroup> =>
  decode.shape({
    contentType: decode.at(
      ["sys", "contentType", "sys", "id"],
      decode.value("paradeGroup")
    ),
    id: decode.at(["sys", "id"], decode.string),
    locale: decode.succeed(locale),
    revision: decode.at(["sys", "revision"], decode.number),
    fields: decode.field(
      "fields",
      decode.shape({
        name: decode.at(["name", locale], decode.string),
        facebookUrl: maybeField(locale, "facebookUrl", decode.string),
        twitterUrl: maybeField(locale, "twitterUrl", decode.string),
        websiteUrl: maybeField(locale, "websiteUrl", decode.string)
      })
    )
  });

export default decodeParadeGroup;
