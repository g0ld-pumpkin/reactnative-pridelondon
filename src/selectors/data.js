// @flow
import type { Event, Events } from "../data/event";
import type { FeaturedEvents } from "../data/featured-events";
import type { FieldRef } from "../data/field-ref";
import type { Performance, Performances } from "../data/performance";
import type { Amenity } from "../data/amenity";
import type { State as DataState } from "../reducers/data";

type ObjectWithId<A> = {
  id: string
} & A;

const reduceToMapHelp = <A>(
  acc: { [id: string]: ObjectWithId<A> },
  item: ObjectWithId<A>
): { [id: string]: ObjectWithId<A> } => {
  acc[item.id] = item; // intentional mutation as this happens in a reduce
  return acc;
};

export const selectLoading = (data: DataState): boolean => data.loading;

export const selectRefreshing = (data: DataState): boolean => data.refreshing;

export const selectEvents = (data: DataState): Array<Event> => data.events;

export const selectEventsMap = (events: Array<Event>): Events =>
  events.reduce(reduceToMapHelp, {});

export const selectEventById = (events: Events, id: string): ?Event =>
  events[id];

export const selectPerformancesMap = (data: DataState): Performances =>
  data.performances;

export const selectPerformanceById = (
  performances: Performances,
  id: string
): ?Performance => performances[id];

export const selectFeaturedEvents = (data: DataState): FeaturedEvents[] =>
  data.featuredEvents;

export const selectFeaturedEventsByTitle = (
  featuredEventsList: FeaturedEvents[],
  title: string
): ?FeaturedEvents =>
  featuredEventsList.find(entry => entry.fields.title === title);

export const selectAmenities = (data: DataState): Array<Amenity> =>
  data.amenities;

const resolveEventsHelp = (eventMap: Events) => (
  acc: Event[],
  reference: FieldRef
) => {
  const event = eventMap[reference.sys.id];
  if (event) {
    acc.push(event);
  }
  return acc;
};

export const resolveEvents = (
  eventMap: Events,
  references: FieldRef[]
): Event[] => references.reduce(resolveEventsHelp(eventMap), []);
