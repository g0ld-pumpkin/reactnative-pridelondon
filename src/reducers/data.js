// @flow
import R from "ramda";
import type { DataAction } from "../actions/data";
import type { Event } from "../data/event";
import type { FeaturedEvents } from "../data/featured-events";
import type { HeaderBanner } from "../data/header-banner";
import type { Images } from "../data/image";
import type { ParadeGroup } from "../data/parade-group";
import type { Performances } from "../data/performance";
import type { Sponsor } from "../data/sponsor";
import type { Amenity } from "../data/amenity";
import { decodeEvent, expandRecurringEvents } from "../data/event";
import decodeFeaturedEvents from "../data/featured-events";
import decodeHeaderBanner from "../data/header-banner";
import { decodeImageDetails } from "../data/image";
import decodeParadeGroup from "../data/parade-group";
import decodePerformance from "../data/performance";
import decodeSponsor from "../data/sponsor";
import decodeAmenity from "../data/amenity";
import locale from "../data/locale";
import type { Decoder } from "../lib/decode";
import { filterMap as decodeFilterMap, map as decodeMap } from "../lib/decode";
import { withDefault as resultWithDefault } from "../lib/result";

export type State = {
  events: Event[],
  featuredEvents: FeaturedEvents[],
  headerBanners: HeaderBanner[],
  images: Images,
  paradeGroups: ParadeGroup[],
  performances: Performances,
  sponsors: Sponsor[],
  amenities: Amenity[],
  loading: boolean,
  refreshing: boolean,
  noDataReceived: boolean
};

const defaultState = {
  events: [],
  featuredEvents: [],
  headerBanners: [],
  images: {},
  paradeGroups: [],
  performances: {},
  sponsors: [],
  amenities: [],
  loading: true,
  refreshing: false,
  noDataReceived: false
};

type ObjectWithId<A> = {
  id: string
} & A;

const reduceToMapHelp = <A>(
  acc: { [id: string]: ObjectWithId<A> },
  item: ObjectWithId<A>
): { [id: string]: ObjectWithId<A> } => {
  acc[item.id] = item; // intentional mutation as this happens in a reduce
  return acc;
};

// moving locale here so we can deal with it in a single place
// this can be moved inside the reducer function if we later want
// to make this dynamic
const decodeEvents: Decoder<Array<Event>> = decodeMap(
  events => R.unnest(events.map(expandRecurringEvents)),
  decodeFilterMap(decodeEvent(locale))
);

const decodeFeaturedEventsCollection: Decoder<
  Array<FeaturedEvents>
> = decodeFilterMap(decodeFeaturedEvents(locale));

const decodeHeaderBanners: Decoder<Array<HeaderBanner>> = decodeFilterMap(
  decodeHeaderBanner(locale)
);

const decodeImages: Decoder<Images> = decodeMap(
  images => images.reduce(reduceToMapHelp, {}),
  decodeFilterMap(decodeImageDetails(locale))
);

const decodeParadeGroups: Decoder<Array<ParadeGroup>> = decodeFilterMap(
  decodeParadeGroup(locale)
);

const decodePerformances: Decoder<Performances> = decodeMap(
  performances => performances.reduce(reduceToMapHelp, {}),
  decodeFilterMap(decodePerformance(locale))
);

const decodeSponsors: Decoder<Array<Sponsor>> = decodeFilterMap(
  decodeSponsor(locale)
);

const decodeAmenities: Decoder<Array<Amenity>> = decodeFilterMap(
  decodeAmenity(locale)
);

const reducer = (state: State = defaultState, action: DataAction) => {
  switch (action.type) {
    case "REQUEST_CMS_DATA":
      return {
        ...state,
        loading: true,
        refreshing: false
      };
    case "REQUEST_UPDATE_CMS_DATA":
      return {
        ...state,
        loading: false,
        refreshing: true
      };
    case "RECEIVE_CMS_DATA":
      return {
        loading: false,
        refreshing: false,
        noDataReceived: false,
        events: resultWithDefault([], decodeEvents(action.data.entries)),
        featuredEvents: resultWithDefault(
          [],
          decodeFeaturedEventsCollection(action.data.entries)
        ),
        headerBanners: resultWithDefault(
          [],
          decodeHeaderBanners(action.data.entries)
        ),
        images: resultWithDefault({}, decodeImages(action.data.assets)),
        paradeGroups: resultWithDefault(
          [],
          decodeParadeGroups(action.data.entries)
        ),
        performances: resultWithDefault(
          {},
          decodePerformances(action.data.entries)
        ),
        sponsors: resultWithDefault([], decodeSponsors(action.data.entries)),
        amenities: resultWithDefault([], decodeAmenities(action.data.entries))
      };
    case "NO_DATA_RECEIVED":
      return {
        ...state,
        loading: false,
        refreshing: false,
        noDataReceived: true
      };
    default:
      return state;
  }
};

export default reducer;
