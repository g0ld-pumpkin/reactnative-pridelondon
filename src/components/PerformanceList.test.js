// @flow
import React from "react";
import { shallow } from "enzyme";
import { generatePerformance, sampleArrayOf } from "../data/__test-data";
import type { Performance } from "../data/performance";
import PerformanceList from "./PerformanceList";

it("renders correctly", () => {
  const performances: Performance[][] = [
    sampleArrayOf(generatePerformance)(2),
    sampleArrayOf(generatePerformance)(2)
  ];
  const output = shallow(<PerformanceList performancePeriods={performances} />);
  expect(output).toMatchSnapshot();
});
