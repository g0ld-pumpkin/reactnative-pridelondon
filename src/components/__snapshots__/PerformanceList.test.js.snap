// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Component>
  <ContentPadding
    padding={Object {}}
    style={Object {}}
  >
    <Text
      accessibilityTraits={
        Array [
          "header",
        ]
      }
      color="lightNavyBlueColor"
      markdown={false}
      markdownStyle={Object {}}
      style={
        Object {
          "marginBottom": 8,
        }
      }
      type="h2"
    >
      Schedule
    </Text>
  </ContentPadding>
  <LayoutColumn
    spacing={20}
  >
    <Component
      key="Afternoon"
    >
      <ContentPadding
        padding={Object {}}
        style={
          Object {
            "backgroundColor": "#f3f3f3",
            "borderWidth": 0,
            "elevation": 3,
            "height": 40,
            "justifyContent": "center",
            "paddingTop": 8,
            "shadowColor": "rgba(0, 0, 0, 0.14)",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 3,
          }
        }
      >
        <Text
          color="lightNavyBlueColor"
          markdown={false}
          markdownStyle={Object {}}
          type="h3"
        >
          Afternoon
        </Text>
      </ContentPadding>
      <ContentPadding
        padding={Object {}}
        style={Object {}}
      >
        <Collapsible
          maxHeight={140}
          showLessLabel="Show less"
          showMoreLabel="Show more"
        >
          <Component
            key="2018-08-08T15:45+00:00"
          >
            <Performance
              startTime="2018-08-08T15:45+00:00"
              title="title"
            />
            <SectionDivider />
          </Component>
          <Component
            key="2018-07-07T06:40+00:00"
          >
            <Performance
              startTime="2018-07-07T06:40+00:00"
              title="title"
            />
          </Component>
        </Collapsible>
      </ContentPadding>
    </Component>
    <Component
      key="Afternoon"
    >
      <ContentPadding
        padding={Object {}}
        style={
          Object {
            "backgroundColor": "#f3f3f3",
            "borderWidth": 0,
            "elevation": 3,
            "height": 40,
            "justifyContent": "center",
            "paddingTop": 8,
            "shadowColor": "rgba(0, 0, 0, 0.14)",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 1,
            "shadowRadius": 3,
          }
        }
      >
        <Text
          color="lightNavyBlueColor"
          markdown={false}
          markdownStyle={Object {}}
          type="h3"
        >
          Afternoon
        </Text>
      </ContentPadding>
      <ContentPadding
        padding={Object {}}
        style={Object {}}
      >
        <Collapsible
          maxHeight={140}
          showLessLabel="Show less"
          showMoreLabel="Show more"
        >
          <Component
            key="2018-08-08T15:45+00:00"
          >
            <Performance
              startTime="2018-08-08T15:45+00:00"
              title="title"
            />
            <SectionDivider />
          </Component>
          <Component
            key="2018-07-07T06:40+00:00"
          >
            <Performance
              startTime="2018-07-07T06:40+00:00"
              title="title"
            />
          </Component>
        </Collapsible>
      </ContentPadding>
    </Component>
  </LayoutColumn>
</Component>
`;
